### DISCUSSION TITLE
ViewComponents with CMS

### DISCUSSION THREAD

**Discussion by Harsh2016** [OrchardCMS/OrchardCore#8381] [Q&A]
Hi,
I have some ViewComponents and I don't want to write the code again for them. Is there any way I can take advantage of CMS  and use existing ViewComponents in my layout.liquid or content.liquid 

Thanks

**Comment by ns8482e** [comment]
Yes you can use ViewComponents in OrchardCore CMS.

There is sample code available in Demo Module 

- Define your view component
https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Demo/Components/FakeViewComponent.cs
- Define razor page for your view component
https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Demo/Views/Shared/Components/Fake/Default.cshtml

- and Invoke in Razor page. 
```
await Component.InvokeAsync("Fake", new { value = i.ToString() })
```

**Reply by Harsh2016** [reply]
Thank you! It resolved my issue, only I needed to use Razor instead of liquid.
